conans/test/integration/toolchains/env/test_virtualenv_powershell.py

```diff
@@ -114,3 +114,53 @@ def test(self):
     assert "MYPS1!!!!" in client.out
     assert "MYVC_CUSTOMVAR1=PATATA1" in client.out
     assert "MYVC_CUSTOMVAR2=PATATA2" in client.out
+
+@pytest.mark.skipif(platform.system() != "Windows", reason="Requires Windows powershell")
+def test_vcvars():
+    client = TestClient()
+    conanfile = textwrap.dedent(r"""
+        from conan import ConanFile
+        from conan.tools.cmake import CMake, cmake_layout, CMakeToolchain, CMakeDeps
+
+        class Conan(ConanFile):
+           settings = "os", "compiler", "build_type", "arch"
+
+           test_requires = [("gtest/1.12.1")]
+
+           generators = 'CMakeDeps', 'CMakeToolchain'
+
+           def layout(self):
+              cmake_layout(self)
+
+           def build(self):
+              cmake = CMake(self)
+              cmake.configure()
+              cmake.build()
+    """)
+    client.save({"conanfile.py": conanfile})
+    hello_cpp = textwrap.dedent(r"""
+        #include "gtest/gtest.h"
+
+        int main(int argc, char **argv) {
+          ::testing::InitGoogleTest(&argc, argv);
+          return RUN_ALL_TESTS();
+        }
+    """)
+    client.save({"hello.cpp": hello_cpp})
+    cmakelists = textwrap.dedent(r"""
+        cmake_minimum_required(VERSION 3.26)
+
+        project(hello-world LANGUAGES CXX)
+        message("C++ compiler: ${CMAKE_CXX_COMPILER}")
+
+        find_package(GTest REQUIRED)
+
+        add_executable(hello hello.cpp)
+        target_link_libraries(hello PRIVATE GTest::gtest GTest::gtest_main)
+    """)
+    client.save({"CMakeLists.txt": cmakelists})
+    client.run("install . -c tools.env.virtualenv:powershell=True -c tools.cmake.cmaketoolchain:generator=Ninja -s compiler.version=193")
+    conanbuild = client.load(r".\build\Release\generators\conanbuild.ps1")
+    vcvars_ps1 = client.load(r".\build\Release\generators\conanvcvars.ps1")
+    assert "conanvcvars.ps1" in conanbuild
```

Comment:
I think a full building example that shows everything fully works would be necessary.

Reply:
I added the build command but this only tests something that was already working before. 
The original issue appeared when activating the environment manually, which I'm not sure if it can be done in the test environment.

Reply:
Have a look to how the powershell is launched, and try to do in the test something like ``powershell ....ps1 ; do-something-to-check``